tests/gha-run-k8s-common.sh

```diff
@@ -91,10 +91,8 @@ function enable_cluster_approuting() {
 
 function install_azure_cli() {
 	curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
-
-	# TODO: Remove the version spec when this bug is fixed:
-	# https://github.com/Azure/azure-cli/issues/31345
-	az extension add --name aks-preview --version 14.0.0b3
+	# The aks-preview extension is required while the Mariner Kata host is in preview.
```

Comment:
We can remove this per my original commit message:
```
Note that I removed the comment about this being a Mariner requirement,
as aks-preview is also a requirement of AKS App Routing, which will
be introduced soon in https://github.com/kata-containers/kata-containers/pull/11164.
```
```suggestion
```

Reply:
Thanks. Amended the commit to keep the commit message compliant.